packages/performance-tests/test/TransformerRegression.test.ts

```diff
@@ -0,0 +1,135 @@
+import { IModelsClient } from "@itwin/imodels-client-authoring";
+import { TestBrowserAuthorizationClient } from "@itwin/oidc-signin-tool";
+import { Reporter } from "@itwin/perf-tools";
+
+async function sleep(ms: number) {
+  return new Promise((resolve) => {
+    setTimeout(resolve, ms);
+  });
+}
+const testCasesMap = new Map([
+  ["identity transform", require("./identity-transformer")],
+]);
+
+const loggerCategory = "Transformer Performance Tests Identity";
+const outputDir = path.join(__dirname, ".output");
+
+const setupTestData = async () => {
+  const logLevel = +(process.env.LOG_LEVEL ?? "2");
+  if (LogLevel[logLevel] !== undefined) {
+    Logger.initializeToConsole();
+    Logger.setLevelDefault(LogLevel.Error);
+    Logger.setLevel(loggerCategory, LogLevel.Info);
+    Logger.setLevel(TransformerLoggerCategory.IModelExporter, logLevel);
+    Logger.setLevel(TransformerLoggerCategory.IModelImporter, logLevel);
+    Logger.setLevel(TransformerLoggerCategory.IModelTransformer, logLevel);
+  }
+  let usrEmail;
+  let usrPass;
+  if(process.env.V2_CHECKPOINT_USER_NAME){
+    usrEmail = process.env.V2_CHECKPOINT_USER_NAME;
+    usrPass = process.env.V2_CHECKPOINT_USER_PASSWORD;
+  } else if(process.env.V1_CHECKPOINT_USER_NAME){
+    usrEmail = process.env.V1_CHECKPOINT_USER_NAME;
+    usrPass = process.env.V1_CHECKPOINT_USER_PASSWORD;
+  } else {
+    usrEmail = process.env.IMODEL_USER_NAME;
+    usrPass = process.env.IMODEL_USER_PASSWORD;
+  }
+
+  assert(usrEmail, "user name was not configured");
+  assert(usrPass, "user password was not configured");
+  const user = {
+    email: usrEmail,
+    password: usrPass,
+  };
+
+  assert(process.env.OIDC_CLIENT_ID, "");
+  assert(process.env.OIDC_REDIRECT, "");
+  assert(process.env.IMJS_URL_PREFIX, "")
+  assert(process.env.OIDC_SCOPE, "List scopes");
+  const authClient = process.env.CI === "1"
+    ? new TestBrowserAuthorizationClient({
+      clientId: process.env.OIDC_CLIENT_ID,
+      redirectUri: process.env.OIDC_REDIRECT,
+      scope: process.env.OIDC_SCOPE,
+      authority: `https://${process.env.IMJS_URL_PREFIX}ims.bentley.com`,
+    }, user)
+    : new NodeCliAuthorizationClient({
+      clientId: process.env.OIDC_CLIENT_ID,
+      redirectUri: process.env.OIDC_REDIRECT,
+      scope: process.env.OIDC_SCOPE,
+    });
+
+  await authClient.signIn();
+
+  const hostConfig  = new IModelHostConfiguration();
+  hostConfig.authorizationClient = authClient;
+  const hubClient = new IModelsClient({ api: { baseUrl: `https://${process.env.IMJS_URL_PREFIX}api.bentley.com/imodels` } });
+  hostConfig.hubAccess = new BackendIModelsAccess(hubClient);
+  await IModelHost.startup(hostConfig);
+
+  return realizeAsyncIterator(getTestIModels());
+};
+
+// Mocha tests must know the test cases ahead time, so we collect the the Imodels first before beginning the tests
+async function realizeAsyncIterator(getTestIModels:AsyncGenerator<TestIModel>) {
```

Comment:
make this function generic:

```ts
async function realizeAsyncIterator<T>(iter: AsyncIterator<T>): T[] {
  //...
```

That type is probably not exactly right, feel free to let me know if you have issues using that. Some permutation of async iterator types should accept the argument `realizeAsyncIterator(myGenerator())`

Reply:
I think AsyncGenerators should be compatible with AsyncIterators but we can defer fixing that until we use this utility in more places, don't worry about it